packages/@aws-cdk/aws-servicecatalog/lib/private/validation.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Class to validate that inputs match requirements.
+ */
+export class InputValidator {
+
+  /**
+   * Validates length is between allowed min and max lengths.
+   */
+  public static validateLength(resourceName: string, inputName: string, minLength: number, maxLength: number, inputString?: string): void {
+    if (inputString !== undefined && (inputString.length < minLength || inputString.length > maxLength)) {
+      throw new Error(
+        `Invalid ${inputName} length of ${inputString.length} - ${inputName} must have a character length between ${minLength} and ${maxLength}\nfor resource ${resourceName}\n`);
```

Comment:
Please include the value that caused the exception to be thrown in the error message.

I would also work a little on the wording of the message here - seems too long to me.